utils/resources.py

```diff
@@ -30,8 +30,11 @@ def write_new_id(filename: str, new_id: int) -> None:
         f_id.write(str(new_id))
 
 def write_cached_announcements(filename: str, cached_announcements: List[int]) -> None:
+    text = ''
+    for cache_id in cached_announcements:
+        text += str(cache_id) + '\n'
```

Comment:
may............ you could make it inline
```python
text = '\n'.join([ str(cache_id) for cache_id in cached_announcements ])
```

Reply:
not sure if that str(cache_id) is really necessary, you could try something like
```python
text = '\n'.join(cached_announcements)
``

Reply:
I prefer the latter, but since cached_announcements is a list of int a conversion to str is needed before performing the string join, I opted to used map as see in the last commit